SmartDeviceLink/SDLRadioControlData.h

```diff
@@ -41,6 +41,37 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (instancetype)initWithFrequencyInteger:(nullable NSNumber<SDLInt> *)frequencyInteger frequencyFraction:(nullable NSNumber<SDLInt> *)frequencyFraction band:(nullable SDLRadioBand)band hdChannel:(nullable NSNumber<SDLInt> *)hdChannel radioEnable:(nullable NSNumber<SDLBool> *)radioEnable hdRadioEnable:(nullable NSNumber<SDLBool> *)hdRadioEnable;
 
+/// Constructs a newly allocated SDLRadioControlCapabilities object with given parameters.
+///
+/// @param frequencyInteger Must be between 875 and 1080
+/// @param frequencyFraction Must be between 0 and 9
+/// @return An instance of the SDLRadioControlData class
+- (instancetype)initFMWithFrequencyInteger:(nullable NSNumber<SDLInt> *)frequencyInteger frequencyFraction:(nullable NSNumber<SDLInt> *)frequencyFraction;
+
+/// Constructs a newly allocated SDLRadioControlCapabilities object with given parameters.
+///
+/// @param hdChannel Must be between 0 and 7
+/// @return An instance of the SDLRadioControlData class
+- (instancetype)initFMWithHdChannel:(nullable NSNumber<SDLInt> *)hdChannel;
```

Comment:
It appears based on this logic, https://github.com/smartdevicelink/sdl_hmi/blob/cd3da85cdb7240a7b5f88d016dea7604158642b1/app/model/media/AudioModels/RadioModel.js#L1493 if provided it will fail if a (frequency or integer) and hdchannel is set, its either one or the other but not both

Reply:
according to the core team "because it's not going to work on some head units, I wouldn't send the `hdchannel` and frequency ever in the same message"

Reply:
But that's fine because they can set it to nil.

Reply:
but according to the bug the reason we are making new inits is to avoid a crash when acidentally setting to many properties. If that is the case what is wrong with using the original inits?

Reply:
The original inits have dozens of properties, most of which cannot be set in order to set data, as described [here](https://smartdevicelink.com/en/guides/iOS/other-sdl-features/remote-control-vehicle-features/#radio) under the column "type".

Reply:
fixed

Reply:
fixed